lemonbar.c

```diff
@@ -1357,9 +1362,11 @@ main (int argc, char **argv)
                 else break;                         // ...bail out
             }
             if (pollin[0].revents & POLLIN) { // New input, process it
-                if (fgets(input, sizeof(input), stdin) == NULL)
-                    break; // EOF received
-
+                *input = 0;
+                while (fgets(input, sizeof input, stdin)) { }
+                if (!*input) {
```

Comment:
No need for this, this case is already handled by `parse`. Moreover this way you don't clear the bar when an empty string is read.